### DISCUSSION TITLE
AxiosResponse not found in 'axios'

### DISCUSSION THREAD

**Discussion by gavinthomas-valtech** [axios/axios#5859] [Q&A]
### Describe the bug

I've upgraded from0.27.2 to 1.2 and now 1.4. It seems both 1.x versions don't import `AxiosResponse`

import axios, { AxiosResponse } from 'axios';

i get an error

```
  1:17  error  AxiosResponse not found in 'axios'  import/named
  ```

### Code snippet

```js
import axios, { AxiosResponse } from 'axios';
```

### Axios Version

1.4.0

### Adapter Version

xhr

### Node.js Version

16.17.0

### Additional Library Versions

```bash
Next 12
```

**Comment by abhijeettidke2012** [comment]
Did you get solution? I am having issue.

**Comment by gavinthomas-valtech** [comment]
For some reason i had to enable this setting in eslint

```
'import/named': 0,
```

not sure why it's changed in axios

**Comment by abhijeettidke2012** [comment]
> For some reason i had to enable this setting in eslint
> 
> ```
> 'import/named': 0,
> ```
> 
> not sure why it's changed in axios

I dont see that property in my code. is it possible for you to share screenshot of code(settings)?

**Comment by gavinthomas-valtech** [comment]
This is in `eslintrc.js` under rules

```
module.exports = {
  extends: [
    'next/core-web-vitals',
    'plugin:@typescript-eslint/recommended',
    'plugin:jsx-a11y/recommended',
    'plugin:import/recommended',
    'plugin:unicorn/recommended',
    'plugin:prettier/recommended',
    'plugin:@tanstack/eslint-plugin-query/recommended',
  ],

  rules: {
    'import/named': 0,
    'unicorn/prefer-module': 'off',
    'unicorn/prefer-query-selector': 'off',
    'unicorn/numeric-separators-style': 'off',
    '@typescript-eslint/ban-types': [
      'error',
      ```

**Comment by abhijeettidke2012** [comment]
> This is in `eslintrc.js` under rules
> 
> ```
> module.exports = {
>   extends: [
>     'next/core-web-vitals',
>     'plugin:@typescript-eslint/recommended',
>     'plugin:jsx-a11y/recommended',
>     'plugin:import/recommended',
>     'plugin:unicorn/recommended',
>     'plugin:prettier/recommended',
>     'plugin:@tanstack/eslint-plugin-query/recommended',
>   ],
> 
>   rules: {
>     'import/named': 0,
>     'unicorn/prefer-module': 'off',
>     'unicorn/prefer-query-selector': 'off',
>     'unicorn/numeric-separators-style': 'off',
>     '@typescript-eslint/ban-types': [
>       'error',
>       ```
> ```

Thanks .. this worked. I was using typescript so it is bit different.

**Comment by bramov** [comment]
Stuck with the same issue on 1.4.0. Is there any way to solve this without using 'import/named' eslint rule?

**Comment by dskoda1** [comment]
Bump, is there a way to import AxiosResponse without needing to change eslint rules? I am happy to contribute if a change needs to be made.

**Comment by amorei22** [comment]
I faced the same issue and solved without changing any eslint configuration. I added "type" after "import" and the problem was gone.
![image](https://github.com/axios/axios/assets/125669251/68d8fdbc-9a74-477a-92b1-7839014b5204)

**Comment by ja-ka** [comment]
https://github.com/un-ts/eslint-import-resolver-ts#configuration seems to be another good answer, too.

**Comment by pathak-codes** [comment]
## Solution: Use TypeScript `import type` keyword

This issue occurs because Axios 1.x changed how it exports TypeScript types. The solution depends on your setup:

### **Recommended Solution (for TypeScript projects):**

Add the `type` keyword to your import statement:

```typescript
import type { AxiosResponse, AxiosRequestConfig } from 'axios';
import axios from 'axios';
```

The `type` keyword tells TypeScript that you're importing only types (which are removed at compile time), not runtime values. This prevents ESLint's `import/named` rule from complaining since no actual exports are being imported.

### **Why this happens:**

In Axios 1.x, `AxiosResponse` is only exported as a type, not as a named export. The error occurs when ESLint tries to validate that the named import exists as a runtime value, which it doesn't.

### **Alternative Solutions:**

1. **Disable ESLint rule** (not recommended):
   ```javascript
   // .eslintrc.js
   rules: {
     'import/named': 0,
   }
   ```

2. **Use eslint-import-resolver-ts** for better TypeScript support:
   - Install: `npm install --save-dev eslint-import-resolver-ts`
   - Configure `.eslintrc.js`:
   ```javascript
   settings: {
     'import/resolver': {
       typescript: true,
     }
   }
   ```

### **Summary:**
The cleanest approach is using `import type` – it's TypeScript best practice and requires no config changes.
